### PR TITLE
First implementation of a jupyter nb based tutorial

### PR DESCRIPTION
**Pull request by LukasGold**
Includes many changes to
* osw.express, as those have been rigorously tested during the writing of the tutorial
* osw.data.import_utility as the design had some flaws and was not as easy to use as it could have been
* Documentation and doc strings

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,12 +1,17 @@
 [![PyPI-Server](https://img.shields.io/pypi/v/osw.svg)](https://pypi.org/project/osw/)
 [![DOI](https://zenodo.org/badge/458130867.svg)](https://zenodo.org/badge/latestdoi/458130867)
-[![Coveralls](https://img.shields.io/coveralls/github/OpenSemanticLab/osw-python/main.svg)](https://coveralls.io/r/<USER>/osw)
+[![Coveralls](https://img.shields.io/coveralls/github/OpenSemanticLab/osw-python/main.svg)](https://coveralls.io/r/OpenSemanticLab/osw)
+[![docs](xx.xx)](https://opensemanticlab.github.io/osw-python/)
+![license](https://img.shields.io/github/license/OpenSemanticLab/osw-python.svg)
+[![Pydantic v2](https://img.shields.io/endpoint?url=https://raw.githubusercontent.com/pydantic/pydantic/main/docs/badge/v2.json)](https://pydantic.dev)
```
Comment (simontaurus) [inline README.md:6]: We still use v1 (within v2 package)

**Review by simontaurus — APPROVED** [review]

File: docs/tutorials/basics.ipynb [hunk 2]
```diff
@@ -0,0 +1,1442 @@
...
+    "    \"properties\": {\n",
+    "        \"additional_property\": { \"type\": \"string\" }\n",
+    "    }\n",
+    "}\n",
+    "```\n",
+    "**Generated Python data classes:**\n",
+    "```python\n",
+    "from osw.model.entity import Entity\n",
+    "\n",
+    "class MyClass(Entity):\n",
```
Comment (simontaurus) [inline docs/tutorials/basics.ipynb:140]: The jsonschema slot above does not inherit from Entity

File: src/osw/model/entity.py [hunk 3]
```diff
@@ -100,6 +102,28 @@ class Entity(OswBaseModel):
     attachments: Optional[List[str]] = Field(None, title="File attachments")
     meta: Optional[Meta] = None
 
+    def __init__(self, **data):
```
Comment (simontaurus) [inline src/osw/model/entity.py:105]: Will this code be regenerated by the code generator? I would suggest to move it to OswBaseModel without throwing any Exception (check if label is set => if yes, set name)

**Review by simontaurus — CHANGES_REQUESTED** [review]
see comments

File: src/osw/auth.py [hunk 4]
```diff
@@ -254,6 +255,7 @@ def save_credentials_to_file(
             if self.cred_filepath is None:
                 filepath_ = [Path.cwd() / CREDENTIALS_FN_DEFAULT]
```
[the remark this answers is not in the document]
Comment (LukasGold) [reply src/osw/auth.py:256]: I agree, the mechanism that is present in osw.express. We could therefore move osw.express.FilePathDefault and CredentialsFpDefault to another module that both osw.auth and osw.express, e.g. osw.custom_types

File: src/osw/model/entity.py (hunk 3 above)
Comment (LukasGold) [reply src/osw/model/entity.py:105]: Recomendation implemented in [commit 0bb9ec95](https://github.com/OpenSemanticLab/osw-python/pull/92/commits/0bb9ec9533864d4c4693087062e7bfb8ebec1ae4)

File: src/osw/core.py [hunk 5]
```diff
@@ -775,6 +780,7 @@ class _ApplyOverwriteParam(OswBaseModel):
         namespace: Optional[str]
         meta_category_title: Optional[str]
         meta_category_template_str: Optional[str]
+        remove_empty_strings: Optional[bool] = True
```
[the remark this answers is not in the document]
Comment (LukasGold) [reply src/osw/core.py:783]: Recommendation implemented in [commit 95fe87ef](https://github.com/OpenSemanticLab/osw-python/pull/89/commits/95fe87ef6afa51212038d606fa9a671f74e9a00c)

File: README.md (hunk 1 above)
Comment (LukasGold) [reply README.md]: Fixed in [commit beffddfd](https://github.com/OpenSemanticLab/osw-python/pull/89/commits/beffddfd5b0e471d844e0769da0747f2ee52109e)

File: src/osw/auth.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (LukasGold) [reply src/osw/auth.py:260]: Will be implemented in separate branch

**Pull request merged**